toolkit/tools/isomaker/maker.go

```diff
@@ -175,36 +179,55 @@ func (im *IsoMaker) setUpIsoGrub2Bootloader() {
 	}()
 
 	logger.Log.Debug("Copying EFI modules into efiboot.img.")
-	// Copy Shim (bootx64.efi) and grub2 (grubx64.efi)
-	bootDirPath := filepath.Join(efiBootImgTempMountDir, "EFI", "BOOT")
-	bootx64EfiFilePath := filepath.Join(bootDirPath, "bootx64.efi")
-	im.extractFromInitrdAndCopy(bootx64BootloaderFile, bootx64EfiFilePath)
-	grubx64EfiFilePath := filepath.Join(bootDirPath, "grubx64.efi")
-	im.extractFromInitrdAndCopy(grubx64BootloaderFile, grubx64EfiFilePath)
-
+	// Copy Shim (boot<arch>64.efi) and grub2 (grub<arch>64.efi)
+	if runtime.GOARCH == "arm64" {
```

Comment:
There's a lot of code repetition here. Let's extract these two cases to a single function with arguments for the boot and Grub2 EFI files.